### ISSUE TITLE
I2C frequency way off on FeatherS3

### ISSUE THREAD

**Issue by ruet-blum** [micropython/micropython#9760]
Default frequency on the FeatherS3 is 851063 and when I set it to 100000 it shows 219780. This seems to be way off.

```
MicroPython v1.19.1 on 2022-06-18; FeatherS3 with ESP32-S3
Type "help()" for more information.
>>> 
>>> i2c =I2C(0)
>>> i2c
I2C(0, scl=9, sda=8, freq=851063)
>>> i2c =I2C(0, freq=100_000)
>>> i2c
I2C(0, scl=9, sda=8, freq=219780)
```
Board: https://micropython.org/download/feathers3/

**Comment by robert-hh** [comment]
There is already PR #9434 for fixing this.

**Comment by dpgeorge** [comment]
Fixed by 12f99481518b0ebcb14f00b2323865a845c2a4f1
